### DISCUSSION TITLE
Invoke-PnPSiteDesign not working

### DISCUSSION THREAD

**Discussion by imsb365** [pnp/powershell#1688] [Q&A]
I have created a custom SharePoint online site design. The design includes below actions,

> - Apply theme "My green theme"
> - Set site external sharing setting
> - Add to hub "My-Home"
> - Add "Home-Editors" to "Members"
> - Add "Home-SuperUsers" to "Owners"
> - Add "Everyone except external users" to "Visitors"
> - Update site regional settings
> - Update branding
> - Add content types from hub
> - Create or update list "Site Pages"
> - Add content type "Intranet Page"

Custom content type 'Intranet Page' is created in admin gallery and it should be synched from there to site.

I have written a PowerShell script that should apply above design to site and then the script updates some other settings in site and library that cannot be achieved from design template.

When I try to execute `Invoke-PnPSiteDesign -Identity $siteDesign.Id -WebUrl $siteUrl` the script just moves to next line without any error or warning, but when I refresh the site in browser design is not applied to SharePoint site. I tried it multiple times but still no luck.
However if I apply the design from UI menu 'Apply a site template' the design is applied.

Is there any bug in Invoke-PnPSiteDesign? or I am missing something?

PS. I have PnP.Powershell version 1.9.0

**Comment by BaronSparky** [comment]
Does `Set-PnPTraceLog -On -Level Debug` before running your script show anything?

**Reply by imsb365** [reply]
I tried setting log trace just before Invoke-PnPSiteDesign, nothing is logged in there.
Then I tried removing actions from site design script one by one, and it appears that "addContentTypesFromHub", this is the culprit. All other actions works except this one.

Here is my site script 
```
{
    "$schema": "https://developer.microsoft.com/json-schemas/sp/site-design-script-actions.schema.json",
    "actions": [
		{
		  "verb": "addContentTypesFromHub",
		  "ids": ["0x0101009D1CB255DA76424F860D91F20E6C4118002A50BFCFB7614729B56886FADA02339B00BAD933FC46326D4CB44E95784821F509"]
		}
    ]
}
```

I knew the change in content type publishing architecture. But above action is suppose to sync the specified CT in site.
Strange thing is same site design works from UI, but not from PnP Powershell.

**Reply by sparkitect** [reply]
I'm seeing the same issue in a customer tenant but there it won't even work from UI.
Did you ever find a resolution, @imsb365 ?

**Reply by Pieter-Veenstra** [reply]
I've got the same issue. I've seen quite a few people complain about this on forums. It is always in relation to addContentTypesFromHub. Did  any of you find a solution?

**Comment by sparkitect** [comment]
Does your Content Type include a Managed Metadata column? If so, you probably have to activate the hidden Taxonomy feature on the site first (Feature ID "73EF14B1-13A9-416b-A9B5-ECECA2B0604C").


________________________________

**Reply by Pieter-Veenstra** [reply]
No managed metadata fields. 

One of my content types uses the Document Set feature but that is enabled on the site. The other content types also don't deploy, Even if I take that document set content type out.

All fields are simple fields ( text, date etc).

Also deploying the site script through the SharePoint UI simply works. This seems to be an issue with the APIS
